src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,884 @@
+
+            with xr.open_dataarray(main_filepath) as da:
+                main = da.load()
+            with xr.open_dataarray(raw_filepath) as da:
+                raw = da.load()
+
+            main_merge = xr.concat([main, raw], dim="T")
+
+            # TODO: use these as temporary files rather
+            #  than download then delete
+            raw_filepath.unlink()
+            main_filepath.unlink()
+
+            # Ensuring fill value encodign is properly set in new
+            # merged dataset
+            if "_FillValue" in main_merge.encoding and np.isnan(
+                main_merge.encoding["_FillValue"]
+            ):
+                main_merge.encoding["_FillValue"] = -999
+
+            main_merge.to_netcdf(main_filepath)
+
+    def _get_directory(self, dir: Union[Path, str]) -> Path:
+        """
+        Return ARC2 directory, either in public, processed
+        or raw for a specific ISO3.
+
+        :param dir: Path or string of folder name in public for passing.
+        """
+        directory = Path(
+            DATA_DIR / PUBLIC_DATA_DIR / dir / self.country_iso3 / ARC2_DIR
+        )
+        return directory
+
+    def _get_raw_filepath(self, main: bool) -> Path:
+        """
+        Return filepath to raw ARC2 data for specific x
+        and y bounds. All data stored within a single
+        main file. If `main`, the filepath is not
+        specified for a specific date range and designated
+        with 'main.nc' at the end. Otherwise, end of
+        filepath designated with `self.date_min` and
+        `self.date_max`.
+
+        :param main: Whether or not the filepath is
+            for the main file.
+        :param temp: Whether or not the filepath is
+            for a temp file used for saving.
+        """
+        directory = self._get_directory(RAW_DATA_DIR)
+
+        if main:
+            end = "main"
+        else:
+            end = f"{self.date_min:%d_%b_%Y}_{self.date_max:%d_%b_%Y}"
+
+        filename = (
+            f"arc2_daily_precip_{self.country_iso3}_"
+            f"{self.range_x[0]}_{self.range_x[1]}_"
+            f"{self.range_y[0]}_{self.range_y[1]}_{end}.nc"
+        )
+        return directory / filename
+
+    def _check_main_exists(self):
+        """Check if main file exists
+
+        Used to check if main file exists
+        """
+        fp = self._get_raw_filepath(main=True)
+        if not fp.exists():
+            raise OSError(
+                "Main file does not exist. "
+                "First run `download_data(main=True)`."
+            )
+
+    def load_raw_data(
+        self,
+        raw_filepath: Union[Path, None] = None,
+    ) -> pd.DataFrame:
+        """
+        Convenience function to load raw raster data, squeeze
+        it and write its CRS. The function always accesses
+        the main file if a filepath is not provided.
+
+        :param raw_filepath: Path to raw file to load. If `None`,
+            loads main file.
+        """
```

Comment:
I don't know if this is the best place to do so, but would it be an idea to somewhere add `self` parameters of the data availability between the min and max date. So e.g. if we set the `date_max` to `2022-01-04` but there is only data till `2022-01-01`, it would be helpful to be able to easily retrieve that somehow.

Reply:
Love the idea.